cmd/transcriber/call/transcriber.go

```diff
@@ -22,10 +25,12 @@ type Transcriber struct {
 	client    *client.Client
 	apiClient *model.Client4
 
+	errCh        chan error
 	doneCh       chan struct{}
+	doneOnce     sync.Once
 	liveTracksWg sync.WaitGroup
 	trackCtxs    chan trackContext
-	startTime    time.Time
+	startTime    atomic.Pointer[time.Time]
```

Comment:
Just wondering, why do we need an atomic.Pointer?

Reply:
Because we need to potentially read/write concurrently. The startTime is set when we receive the recording state (meaning it started processing):

https://github.com/mattermost/calls-transcriber/blob/0909fc1dad5b4a8dddabbde8cc52db81676d1569/cmd/transcriber/call/transcriber.go#L99

At the time we could already be receiving data from audio tracks:

https://github.com/mattermost/calls-transcriber/blob/0909fc1dad5b4a8dddabbde8cc52db81676d1569/cmd/transcriber/call/tracks.go#L122-L125

Reply:
Ah, I didn't see that it was being read in tracks. Thanks!